### PR TITLE
Add checks around creation for cert headers

### PR DESCRIPTION
**Pull request by rado17**
Generate certificate headers only if Enterprise is enabled.

### CODE REVIEW

**Comment by jukkar** [comment]
This seems to have relation to #79973 where I am fixing lot of compilation issues related to various Kconfig options.

File: subsys/net/l2/wifi/CMakeLists.txt [hunk 1]
```diff
@@ -23,6 +23,7 @@ if (CONFIG_WIFI_NM)
 zephyr_iterable_section(NAME wifi_nm_instance GROUP DATA_REGION ${XIP_ALIGN_WITH_INPUT} SUBALIGN ${CONFIG_LINKER_ITERABLE_SUBALIGN})
 endif()
 
+if (CONFIG_WIFI_NM_WPA_SUPPLICANT_CRYPTO_ENTERPRISE)
```
Comment (krish2718) [inline subsys/net/l2/wifi/CMakeLists.txt:26]: the block needs to be indented.

**Review by krish2718 — DISMISSED** [review]

**Comment by jukkar** [comment]
@rado17 could you rebase, there is now a conflict

**Review by krish2718 — DISMISSED** [review]

**Review by jukkar — DISMISSED** [review]

File: subsys/net/l2/wifi/CMakeLists.txt [hunk 2]
```diff
@@ -23,65 +23,61 @@ if (CONFIG_WIFI_NM)
...
+      )
 
-generate_inc_file_for_target(
-    app
-    ${ZEPHYR_BASE}/samples/net/wifi/test_certs/server-key.pem
-    ${gen_dir}/server-key.pem.inc
-    )
-# Add explicit dependency on app target for ZEPHYR_CURRENT_LIBRARY, so these
-# headers are generated at the correct point in the build
-add_dependencies(${ZEPHYR_CURRENT_LIBRARY} app)
```
Comment (MaochenWang1) [inline subsys/net/l2/wifi/CMakeLists.txt:85]: Why remove this `add_dependencies(${ZEPHYR_CURRENT_LIBRARY} app)`?
Which can fix the CI build error that xxx.pem.inc included but not generated yet.

**Review by MaochenWang1 — CHANGES_REQUESTED** [review]

File: subsys/net/l2/wifi/CMakeLists.txt (hunk 2 above)
Comment (rado17) [reply subsys/net/l2/wifi/CMakeLists.txt:85]: Removed by mistake. Added it back.

**Review by MaochenWang1 — DISMISSED** [review]

File: subsys/net/l2/wifi/CMakeLists.txt [hunk 3]
```diff
@@ -23,65 +23,65 @@ if (CONFIG_WIFI_NM)
 zephyr_iterable_section(NAME wifi_nm_instance GROUP DATA_REGION ${XIP_ALIGN_WITH_INPUT} SUBALIGN ${CONFIG_LINKER_ITERABLE_SUBALIGN})
 endif()
 
-# Wi-Fi Enterprise test certificates handling
-set(gen_inc_dir ${ZEPHYR_BINARY_DIR}/misc/generated)
-set(gen_dir ${gen_inc_dir}/wifi_enterprise_test_certs)
-# Create output directory for test certs
-file(MAKE_DIRECTORY ${gen_dir})
+if(CONFIG_WIFI_NM_WPA_SUPPLICANT_CRYPTO_ENTERPRISE)
```
Comment (krish2718) [inline subsys/net/l2/wifi/CMakeLists.txt:26]: can you please also add `if CONFIG_NET_L2_WIFI_SHELL` as these are only used by Wi-Fi shell?

Comment (rado17) [reply subsys/net/l2/wifi/CMakeLists.txt]: Done.

**Review by krish2718 — APPROVED** [review]

**Review by MaochenWang1 — APPROVED** [review]

**Pull request merged**